src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1604,11 +1639,22 @@ impl Disambiguator {
         Suggestion::Prefix(prefix)
     }
 
+    fn is_field(self) -> bool {
+        self == Self::Kind(DefKind::Field)
+    }
+
     fn ns(self) -> Namespace {
         match self {
             Self::Namespace(n) => n,
             Self::Kind(k) => {
-                k.ns().expect("only DefKinds with a valid namespace can be disambiguators")
+                match k {
+                    // Fields technically don't have a namespace, but we treat
+                    // fields as if they belong to the value namespace.
+                    DefKind::Field => TypeNS,
```

Comment:
```suggestion
                    // Fields technically don't have a namespace, but we treat
                    // fields as if they belong to the value namespace.
                    DefKind::Field => ValueNS,
```
Do we need to return a namespace here? Would it be better to return an Option, the same way `DefKind::ns` does?

Reply:
Oops... I said they belong to the value namespace because I saw that elsewhere, but I think in my mind I thought type namespace fitted better. Need to figure out if it makes sense to update elsewhere to use `TypeNS` or to switch to using `ValueNS` here...